docs/ethereum_attestation.md

```diff
@@ -0,0 +1,133 @@
+
+Currently supported `format` values:
+- `1` - `payload` is exactly 20 bytes and contains an SHA-1 Radicle Identity root hash
+- `2` - `payload` is exactly 32 bytes and contains an SHA-256 Radicle Identity root hash
+
+We need to deploy an official instance of the `Claims` smart contract and
+it must be used by all the users.
+If anybody makes a claim using a different instance, it won't be recognized by others.
+
+## Creation of an attestation
+
+You need to perform 2 actions in any order:
+- Add or update an `https://radicle.xyz/ethereum/claim/v1` entry in your identity doc.
+The entry's `address` must be your Ethereum address.
+It's highly recommended to set a short expiration date as Ethereum claims don't expire.
+- Call `claim` in the `Claims` smart contract. The `root` must point to your link identity.
+
+## Discovery from an Ethereum address
+
+When you know an Ethereum address, you can find the claimed link identity using an Ethereum client.
+The example calls are based on the standard [client JSON RPC API][rpc] and should be exposed
+by your favourite Ethereum client library.
+It's important that the client must be trusted not to hide the events.
+
+- Use [getLogs][rpc-logs] to get the newest `Claimed` event filtered for the given ethereum address
+- Get the event's `transactionHash` field and use it to fetch the transaction which emitted it with
+[getTransactionByHash][rpc-tx]
+- Validate that the transaction signature matches its data and the ethereum address.
+For reference the signature payload content is listed [here][rpc-sign].
+- Read the Link identity root hash from the transaction data
+- Verify that the Link identity doc claims back the Ethereum address,
+see [Discovery from a Link identity](#discovery-from-a-link-identity)
+
+## Discovery from a Link identity
+
+When you know a Link identity, you can find the claimed Ethereum address.
+Obtain the tip of its identity chain and read the Ethereum address from the identity doc
+`address` field in section `https://radicle.xyz/ethereum/claim/v1`, unless it's expired.
+You need to verify that the given Ethereum address claims back the Link identity root,
+see [Discovery from an Ethereum address](#discovery-from-an-ethereum-address).
+
+## Revocation of an attestation
+
+When your attestation for whatever reason is no longer valid,
+you should revoke it as soon as possible.
+Only one claim needs to be revoked to break the attestation,
+but to improve security you should revoke both sides if you can.
+
+To revoke a claim on Link, update the identity doc and publish it in a new revision.
+You can change the claimed Ethereum address or remove
+the `https://radicle.xyz/ethereum/claim/v1` section altogether.
+The other Link nodes will not notice this update until they fetch the new revision.
+
+To revoke a claim on Ethereum, call the `claim` function in `Claims` contract.
+You can claim a different Link identity root or the `0` root to revoke any claim you may have.
```

Comment:
These two sentences seem disconnected. Reading the first seem that just calling `claim` will be enough to revoke, whereas in the seconds it starts by talking about claiming another identity.

Reply:
Hmm, the first sentence describes the action and the second its details, `claim` can't be called without any parameters. Do you have an idea for a clearer expression of that?

Reply:
It's good now :ok_hand: